File: go/behavioral/state/state_test.go

package state

import (
	"testing"
	"../../lib/asserts"
)

func TestState (t *testing.T) {

		ctx := NewDvdStateContext()

		asserts.Equals( t, "State 1 test", 
			ctx.ShowName( "Sponge Bob Squarepants - Nautical Nonsense and Sponge Buddies"),
			"Sponge*Bob Squarepants - Nautical Nonsense and Sponge Buddies")

		asserts.Equals( t, "State 2 test", 
			ctx.ShowName( "Jay and Silent Bob Strike Back"),
			 "Jay*and*Silent Bob Strike Back")

		asserts.Equals( t, "State 3 test", 
			ctx.ShowName( "Buffy The Vampire Slayer Season 2"),
			 "Buffy!The Vampire Slayer Season 2")

		asserts.Equals( t, "State 4 test", 
			ctx.ShowName( "The Sopranos Season 2"),
			 "The*Sopranos Season 2")
		
}      

File: clojure/patterns/test/patterns/behavioral/state_test.clj
(ns patterns.behavioral.state-test
  (:require 
		[clojure.test :refer :all]
      [patterns.behavioral.state :refer :all]
	))

(deftest a-test
	(let [c (circle 1)]
		(testing "First test"
			(is (= (:rd c) (:rd c))))))
(comment
(deftest a-test
	(let [ctx dvdstatecontext.])
		(testing "First show name"
			(is (= "Sponge*Bob Squarepants - Nautical Nonsense and Sponge Buddies" (show-name ctx "Sponge Bob Squarepants - Nautical Nonsense and Sponge Buddies"))))

		(testing "Second show name"
			(is (= "Jay*and*Silent Bob Strike Back" (show-name ctx "Jay and Silent Bob Strike Back"))))

		(testing "Third show name"
			(is (= "Buffy!The Vampire Slayer Season 2" (show-name ctx "Buffy The Vampire Slayer Season 2"))))

		(testing "Fourth show name"
			(is (= "The*Sopranos Season 2" (show-name ctx "The Sopranos Season 2"))))
))
